test/assert.js

```diff
@@ -648,6 +652,17 @@ test('.throws() fails if passed a bad value', t => {
 	t.end();
 });
 
+test('promise .throws() fails when promise is resolved', t => {
+	return assertions.throws(Promise.resolve('foo'))
+	.catch(err => {
+		testFailResults(t, err, {
```

Comment:
Rather than extracting the comparison logic into `testFailResults`, if you capture `lastFailure` through the `pending()` function you could write an `eventuallyFailsWith()`:

```js
function eventuallyFailsWith(t, promise, subset) {
  return promise.then(() => failsWith(t, () => {}, subset))
}
```

Given that `assertion.throws()` should return a promise that always resolves, and the error will be captured separately.